phpBB/phpbb/ban/type/email.php

```diff
@@ -28,7 +28,7 @@ public function get_type(): string
 	/**
 	 * {@inheritDoc}
 	 */
-	public function get_user_column(): ?string
+	public function get_user_column(): string|null
```

Comment:
I guess it's more readable this way but was there another reason you went with this format?

Reply:
I would kind-of prefer the `?string` over the `string|null` version just because it better resembles how we deal with this in the config.yml too.

Also a question, don't any of these null params that appear in class constructors need to be done in the config.yml files too? For example in `phpBB/phpbb/composer/installer.php` if `config|null $config = null)` then shouldn't `'@config'` in this class's definition in the config yaml be `'@?config'`

Reply:
> I guess it's more readable this way but was there another reason you went with this format?

Nope, also it's just for consistency as different syntaxes were used in different places.

Reply:
>  Or am I just confusing this nullable stuff with optional args?

Probably yes, it can be passed as null explicitly.

Reply:
I have proposed this change to the coding guidelines for 4.0:
https://github.com/phpbb/documentation/pull/299

Reply:
Then code sniffer should check that probably.

Reply:
> Then code sniffer should check that probably.

May be in another ticket/PR? This requires more work to dig into custom sniffs, couldn't find ready ones.

Reply:
Sure, fine with me